### ISSUE TITLE
Document from webpage different when loaded inside or outside of method

### ISSUE THREAD

**Issue by THoferSchmitz** [AngleSharp/AngleSharp#906]
Hi,

I'm trying to load a webpage with AngleSharp. My code looks like this:
```
void MyFunction()
{
    forach( string url in GetUrls() )
    {
        MyData = ParseUrl(url);
    }
}

MyData ParseUrl(string url)
{
    var config = Configuration.Default.WithDefaultLoader();
    var context = BrowsingContext.New(config);
    IDocument document;
    do
    {
        var task = context.OpenAsync(url);
        task.Wait();
        document = task.Result;
    } while (document.Body.ChildElementCount == 0);
    // search for classes and attributes here
}
```
I had to load the page in a loop because otherwise I sometimes got an empty document. This code works fine, the html looks just like in my browser.

Now I wanted to load the document ouside of my parse function and pass the document instead of the url string, like this:
```
void MyFunction()
{
    var config = Configuration.Default.WithDefaultLoader();
    var context = BrowsingContext.New(config);
    forach( string url in GetUrls() )
    {
        IDocument document;
        do
        {
            var task = context.OpenAsync(url);
            task.Wait();
            document = task.Result;
        } while (document.Body.ChildElementCount == 0);
        MyData = ParseUrl(document);
    }
}

MyData ParseUrl(IDocument document)
{
    // search for classes and attributes here
}
```
But moving the loading part outside the function (with no other changes) causes my parse method to fail because the elements I searched for could no longer be found. I printed the documents html and it's very different from what is in my browser.

I tried moving the code back and forth several times without changing anything else. It's always the same result/error.

**Comment by FlorianRappl** [comment]
Before going into details it is quite hard to follow if you cannot come up with some HTML / page for making this reproducible.

Furthermore, for me the code above does not make much sense - for instance, the use `Wait()` instead of just using `Result` directly if you really want to de-async the code (which you should not) - is questionable.

When you refer to "HTML in the browser" I assume you do not mean "The HTML that the browser sees", but rather "The DOM as I can inspect in the browser". For the latter, in any case make sure that there are no JS fragments making the change. Otherwise, it cannot be compared.

The report also misses very critical information like the used version of AngleSharp or your runtime.

I'd leave it open for now hoping that this becomes somewhat reproducible / filled with all relevant information.
